Content.Server/GameObjects/Components/Timing/UseDelayComponent.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using CancellationTokenSource = System.Threading.CancellationTokenSource;
+using Content.Shared.GameObjects.Components.Items;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Serialization;
+using Robust.Shared.Timers;
+using Robust.Shared.IoC;
+using Robust.Shared.Interfaces.Timing;
+
+namespace Content.Server.GameObjects.Components.Timing
+{
+    /// <summary>
+    /// Timer that creates a cooldown each time an object is activated/used
+    /// </summary>
+    [RegisterComponent]
+    public class UseDelayComponent : Component
+    {
+        public override string Name => "UseDelay";
+
+        private TimeSpan _lastUseTime;
+
+        /// <summary>
+        /// The time, in milliseconds, between an object's use and when it can be used again
+        /// </summary>
+        private int _delay;
+        public int Delay { get => _delay; set => _delay = value; }
```

Comment:
I'd rather have this delay stored in seconds as a float actually, come to think of it.

Reply:
Timer.Spawn doesn't seem to take floats

Reply:
Use `TimeSpan.FromSeconds`